src/lib/db-types.ts

```diff
@@ -185,3 +188,21 @@ export interface DBOfferingGroupUser {
   connectedTimestamp: number;
   disconnectedTimestamp?: number;
 }
+
+export enum DBSectionType {
+  all = "all"
+}
+export type DBSupportSectionTarget = SectionType | DBSectionType;
```

Comment:
I did end up using the idea that an "all"-target support is equivalent to a "problem" support for the purposes of MST, so I kept the "all" notion in DBTypes. The "audience" type is used both in Firebase and MST, so I pulled that type out into the supports model.

Also I used string enums as types in this PR and it went pretty well!
Also I made